src/Pact/Analyze/Check.hs

```diff
@@ -263,31 +265,57 @@ moduleTables modules (_mod, modRefs) = do
 
     let schemaName = asString (_utName schema)
 
-    -- look through every meta-property in the schema for invariants
-    let mExp :: Maybe Exp
-        mExp = schemas ^? ix schemaName.tMeta._Just.mMetas.ix "invariants"
+        invariants = schemas ^? ix schemaName.tMeta._Just.mMetas.ix "invariants"
+        invariant  = schemas ^? ix schemaName.tMeta._Just.mMetas.ix "invariant"
 
-    let invariants :: [(Text, SchemaInvariant Bool)]
-        invariants = case mExp of
-          Just (Pact.EList' exps) -> catMaybes $ flip fmap exps $ \meta -> do
-            SomeSchemaInvariant expr TBool
-              <- expToInvariant (_utFields schema) meta
-            [v] <- pure $ Set.toList (invariantVars expr)
-            pure (v, expr)
-          _                  -> []
+        exps :: [Exp]
+        exps = case invariants of
+          Just (Pact.ELitList x) -> x
+          -- TODO(joel): don't just throw an error
+          Just _ -> error "invariants must be a list"
```

Comment:
Is it currently possible for an end-user to write concrete syntax that would trigger this case? or are we protected by the parser?

Reply:
`(invariants (> x 5)` :(

Reply:
We should probably address this one before merging then

Reply:
Erroring is much better than what it currently does. Fixing this properly will still show an error message -- just a prettier one. I just want to make sure we get this in the release.

Reply:
Sounds good. I'm refactoring other stuff at the moment that touches a lot of the same code anyhow